### PR TITLE
Api refactor

### PR DESCRIPTION
**Pull request by fwpushan**
1. Add new version management support for Schema
2. Add new generic csv data load logic for schema
3. Re organize   generated sql file dir structure.
4. Update tests

### CODE REVIEW

File: api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts [hunk 1]
```diff
@@ -60,16 +60,16 @@ export class CreateObservationCode1562358560325 implements MigrationInterface {
...
-        await queryRunner.query(getSQLFileData(this.specificUseCodeSchema.dataSQLPath()));
-        await queryRunner.query(getSQLFileData(this.slopeCodeSchema.dataSQLPath()));
-        await queryRunner.query(getSQLFileData(this.aspectCodeSchema.dataSQLPath()));
-        await queryRunner.query(getSQLFileData(this.proposedCodeSchema.dataSQLPath()));
+        await queryRunner.query(getSQLFileData(this.densitySchema.dataSQLPath(), this.densitySchema.className));
+        await queryRunner.query(getSQLFileData(this.distributionSchema.dataSQLPath(), this.distributionSchema.className));
+        await queryRunner.query(getSQLFileData(this.agencyCodeSchema.dataSQLPath(), this.agencyCodeSchema.className));
+        await queryRunner.query(getSQLFileData(this.observationTypeCodeSchema.dataSQLPath(), this.observationTypeCodeSchema.className));
+        await queryRunner.query(getSQLFileData(this.soilTextureCodeSchema.dataSQLPath(), this.soilTextureCodeSchema.className));
+        await queryRunner.query(getSQLFileData(this.observationGeometryCodeSchema.dataSQLPath(), this.observationTypeCodeSchema.className));
```
Comment (andrea-williams) [inline api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts]: shouldn't this be "this.observationGeometryCodeSchema.className" ?

File: api/api_sources/sources/libs/core-database/baseSchema.ts [hunk 2]
```diff
@@ -431,6 +492,22 @@ export class  BaseSchema {
     public static get id(): string {
         return this.shared.table.columns.id;
     }
+
+    /**
+     * @description Get all migration sql file related to schema
+     * @returns sting []
```
Comment (andrea-williams) [inline api/api_sources/sources/libs/core-database/baseSchema.ts]: string (if we want to be really picky...)

File: api/api_sources/sources/libs/core-database/baseSchema.ts [hunk 3]
```diff
@@ -431,6 +492,22 @@ export class  BaseSchema {
...
+     */
+    public get migrationFiles(): {[key: string]: string} {
+        return SchemaHelper.shared.migrationFiles(this);
+    }
+
+    /**
+     * @description Get all revert migration sql file related to schema
+     * @returns sting []
```
Comment (andrea-williams) [inline api/api_sources/sources/libs/core-database/baseSchema.ts]: string (if we want to be really picky...)

File: api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts (hunk 1 above)
Comment (fwpushan) [reply api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts]: Fixed

File: api/api_sources/sources/libs/core-database/baseSchema.ts (hunk 3 above)
Comment (fwpushan) [reply api/api_sources/sources/libs/core-database/baseSchema.ts]: Fixed

File: api/api_sources/sources/libs/core-database/baseSchema.ts (hunk 2 above)
Comment (fwpushan) [reply api/api_sources/sources/libs/core-database/baseSchema.ts]: Fixed

File: api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts (hunk 1 above)
Comment (fwpushan) [reply api/api_sources/sources/database/migrations/1562358560325-CreateObservationCodes.ts]: Fixed

**Pull request merged**